internal/msalbase/id_token.go

```diff
@@ -29,32 +31,45 @@ type IDToken struct {
 	IssuedAt          int64  `json:"iat,omitempty"`
 	NotBefore         int64  `json:"nbf,omitempty"`
 	RawToken          string
+
+	AdditionalFields map[string]interface{}
 }
 
-// CreateIDToken creates an ID token instance from a JWT
-func CreateIDToken(jwt string) (*IDToken, error) {
+// CreateIDToken creates an ID token instance from a JWT.
+func CreateIDToken(jwt string) (IDToken, error) {
 	jwtArr := strings.Split(jwt, ".")
 	if len(jwtArr) < 2 {
-		return nil, errors.New("id token returned from server is invalid")
+		return IDToken{}, errors.New("id token returned from server is invalid")
 	}
 	jwtPart := jwtArr[1]
 	jwtDecoded, err := DecodeJWT(jwtPart)
 	if err != nil {
-		return nil, err
+		return IDToken{}, err
 	}
-	idToken := &IDToken{}
-	err = json.Unmarshal(jwtDecoded, idToken)
+	idToken := IDToken{}
+	err = json.Unmarshal(jwtDecoded, &idToken)
 	if err != nil {
-		return nil, err
+		return IDToken{}, err
 	}
 	idToken.RawToken = jwt
 	return idToken, nil
 }
 
-// GetLocalAccountID extracts an account's local account ID from an ID token
-func (idToken *IDToken) GetLocalAccountID() string {
-	if idToken.Oid != "" {
-		return idToken.Oid
+// IsZero indicates if the IDToken is the zero value.
+func (i IDToken) IsZero() bool {
```

Comment:
Is this a language idiom to avoid using null? The IdToken is optional in some flows, indeed. Is this function supposed to capture this?

Reply:
Some of the code relied on IDToken == nil, as you've indicated.  This allows you to detect the same condition with a non-pointer as you correctly guessed.  You'll see this naming for a method in a few places in stdlib and has enough use to be called an idiom.  In open source code I'd say that overuse of pointers for detection is probably more prevalent.  That is usually a mistake if you want performance, as  it might escape and get put on the heap.

This can be done easier in some cases, like:

```go
type myStruct struct {
  name string
}

var zeroMyStruct = myStruct{}

func (m myStruct) isZero() bool {
  if m == zeroMyStruct {
    return true
  }
  return false
```

But you can't do struct comparisons if you have slices or maps in the struct (sigh).